packages/@n8n/config/src/decorators.ts

```diff
@@ -0,0 +1,79 @@
+import 'reflect-metadata';
+import { readFileSync } from 'fs';
+import { Container, Service } from 'typedi';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type Class = Function;
+type PropertyKey = string | symbol;
+interface PropertyMetadata {
+	type: unknown;
+	envName?: string;
+}
+
+const globalMetadata = new Map<Class, Map<PropertyKey, PropertyMetadata>>();
+
+export const Config: ClassDecorator = (ConfigClass: Class) => {
+	const factory = function () {
+		const config = new (ConfigClass as new () => Record<PropertyKey, unknown>)();
+		const classMetadata = globalMetadata.get(ConfigClass);
+		if (!classMetadata) {
+			// eslint-disable-next-line n8n-local-rules/no-plain-errors
+			throw new Error('Invalid config class: ' + ConfigClass.name);
```

Comment:
Should we enable `n8n-local-rules/no-plain-errors` for this package? In this case it makes no difference as failing here breaks the build and is not reportable, but for future proofing any other errors we may need to throw in this package. Maybe the rule should be live at the top lint config file and be disabled on a case-by-case basis.

Reply:
I'd like to see if we can make the type-checker somehow catch these kind of errors, to prevent stuff like this even reaching runtime.